src/backend/EdFi.DataManagementService.Backend.Postgresql.Test.Integration/DatabaseTest.cs

```diff
@@ -11,271 +12,275 @@
-                            ResourceName = AsValueType<IMetaEdProjectName, string>(resourceName == null ? "" : resourceName),
-                            IsDescriptor = false
-                        }
-                    ).ActLike<IResourceInfo>();
+                ResourceInfo = CreateResourceInfo(resourceName),
+                SearchParameters = searchParameters,
+                PaginationParameters = paginationParameters,
+                TraceId = new TraceId("123")
             }
-            return (
-                new
-                {
-                    resourceInfo = resourseInfo,
-                    searchParameters,
-                    paginationParameters,
-                    TraceId = new TraceId("123")
-                }
-            ).ActLike<IQueryRequest>();
-        }
+        ).ActLike<IQueryRequest>();
+    }
 
-        public static IDeleteRequest CreateDeleteRequest(Guid documentUuidGuid)
-        {
-            return (
-                new
-                {
-                    ResourceInfo = _resourceInfo,
-                    TraceId = new TraceId("123"),
-                    DocumentUuid = new DocumentUuid(documentUuidGuid)
-                }
-            ).ActLike<IDeleteRequest>();
-        }
+    protected static IDeleteRequest CreateDeleteRequest(string resourceName, Guid documentUuidGuid)
+    {
+        return (
+            new
+            {
+                ResourceInfo = CreateResourceInfo(resourceName),
+                TraceId = new TraceId("123"),
+                DocumentUuid = new DocumentUuid(documentUuidGuid)
+            }
+        ).ActLike<IDeleteRequest>();
+    }
 
-        public static UpsertDocument CreateUpsert(NpgsqlDataSource dataSource)
-        {
-            return new UpsertDocument(dataSource, new SqlAction(), NullLogger<UpsertDocument>.Instance);
-        }
+    /// <summary>
+    /// Takes a DB setup function and two DB operation functions. The setup function is given
+    /// a managed connection and transaction and runs on the current thread. Transaction
+    /// commit is provided and no results are returned to the caller.
+    ///
+    /// The two functions which perform DB operations have their transactions orchestrated
+    /// such that the first transaction executes without committing, then the second transaction
+    /// executes and blocks in the DB until the first transaction is committed, at which
+    /// point the second transaction either completes or aborts due to a concurrency problem.
+    ///
+    /// The first operation is given a managed connection and transaction provided
+    /// on the current thread, whereas the second operation has a managed connection and
+    /// transaction provided to it on a separate thread.
+    ///
+    /// This method returns the operation results for each operation performed.
+    ///
+    /// </summary>
+    protected async Task<(T?, U?)> OrchestrateOperations<T, U>(
```

Comment:
This lets us write tests of two requests with overlapping transactions, important to simulate write-conflict scenarios - and test that other scenarios **aren't** write-conflicts.

Reply:
FYI, GitHub's lousy diffing makes some of it hard to read. Ignore the red mixed in.